UPGRADE.md

```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
 configuration, without dropping existing in-flight connections.
 
 
+## Upgrade to `2.1.0`
+
+Kong adheres to [semantic versioning](https://semver.org/), which makes a
+distinction between "major", "minor" and "patch" versions. The upgrade path
+will be different on which previous version from which you are migrating.
+
+If you are migrating from 2.0.0, upgrading into 2.1.x is a minor upgrade,
+but read below for important instructions on database migration, especially
+for Cassandra users.
+
+If you are migrating from 1.x, upgrading into 2.1.x is a major upgrade,
+so, in addition, be aware of any [breaking changes](#breaking-changes-2.0.0)
+between the 1.x and 2.x series below, further detailed in the
+[CHANGELOG.md](https://github.com/Kong/kong/blob/2.0.0/CHANGELOG.md) document.
+
+#### 1. Dependencies
+
+If you are using the provided binary packages, all necessary dependencies
+for the gateway are bundled and you can skip this section.
+
+If you are building your dependencies by hand, there are changes since the
+previous release, so you will need to rebuild them with the latest patches.
+
+In order to use all Kong features, including the new
+dynamic upstream keepalive behavior, the required OpenResty version is
+[1.15.8.3](http://openresty.org/en/changelog-1015008.html) and the
+the set of [OpenResty
+patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools/openresty-patches)
+included has changed, including the latest release of
+[lua-kong-nginx-module](https://github.com/Kong/lua-kong-nginx-module).
+Our [kong-build-tools](https://github.com/Kong/kong-build-tools)
```

Comment:
```suggestion
Our [openresty-build-tools](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools)
```

Reply:
(pointing to `kong-build-tools` root might be a little confusing to users unfamiliar with those tools, as that is broader than just building openresty)

Reply:
This was copied from the previous upgrade tests. Calling it "our openresty-build-tools repository" would be misleading. I'll try a different adjustement!